zstash/utils.py

```diff
@@ -82,7 +93,11 @@ def get_files_to_archive(cache: str, exclude: str) -> List[str]:
         if x[0] != os.path.join(".", cache)
     ]
 
-    # Eliminate files based on exclude pattern
+    # First, add files based on include pattern
+    if include is not None:
+        files = include_files(include, files)
```

Comment:
It is sufficient to determine which files to add to the `zstash archive` after getting the entire list of files, as the archiving step (which comes later) is the computationally expensive part.